spark/src/test/java/com/netflix/iceberg/spark/source/TestSplitPlanning.java

```diff
@@ -0,0 +1,144 @@
+    optional(2, "data", Types.StringType.get())
+  );
+
+  private static SparkSession spark = null;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  private Table table = null;
+  private String tableLocation = null;
+
+  @BeforeClass
+  public static void startSpark() {
+    TestSplitPlanning.spark = SparkSession.builder().master("local").getOrCreate();
+  }
+
+  @AfterClass
+  public static void stopSpark() {
+    SparkSession spark = TestSplitPlanning.spark;
+    TestSplitPlanning.spark = null;
+    spark.stop();
+  }
+
+  @Before
+  public void setupTable() throws IOException {
+    File tableDir = temp.newFolder();
+    this.tableLocation = tableDir.toURI().toString();
+    this.table = TABLES.create(SCHEMA, tableLocation);
+  }
+
+  @Test
+  public void testBasicSplitPlanning() {
+    List<DataFile> files128MB = newFiles(4, 128 * 1024 * 1024);
+    appendFiles(files128MB);
+    Dataset<Row> df1 = spark.read().format("iceberg").load(tableLocation);
+    Assert.assertEquals(4, df1.toJavaRDD().getNumPartitions());
+
+    List<DataFile> files32MB = newFiles(16, 32 * 1024 * 1024);
+    appendFiles(files32MB);
+    Dataset<Row> df2 = spark.read().format("iceberg").load(tableLocation);
+    Assert.assertEquals(8, df2.toJavaRDD().getNumPartitions());
+  }
+
+  @Test
+  public void testSplitPlanningWithSmallFiles() {
+    table.updateProperties().set(TableProperties.SPLIT_LOOKBACK, "30").commit();
+    List<DataFile> files60MB = newFiles(50, 60 * 1024 * 1024);
+    List<DataFile> files5KB = newFiles(370, 5 * 1024);
+    Iterable<DataFile> files = Iterables.concat(files60MB, files5KB);
+    appendFiles(files);
+    Dataset<Row> df = spark.read().format("iceberg").load(tableLocation);
+    Assert.assertEquals(35, df.toJavaRDD().getNumPartitions());
+  }
+
+  @Test
+  public void testSplitPlanningWithNoMinWeight() {
+    table.updateProperties()
+      .set(TableProperties.SPLIT_LOOKBACK, "30")
+      .set(TableProperties.SPLIT_MIN_FILE_WEIGHT, "0")
+      .commit();
+    List<DataFile> files60MB = newFiles(2, 60 * 1024 * 1024);
+    List<DataFile> files5KB = newFiles(100, 5 * 1024);
+    Iterable<DataFile> files = Iterables.concat(files60MB, files5KB);
+    appendFiles(files);
+    Dataset<Row> df = spark.read().format("iceberg").load(tableLocation);
+    Assert.assertEquals(1, df.toJavaRDD().getNumPartitions());
+  }
+
+  private void appendFiles(Iterable<DataFile> files) {
+    AppendFiles appendFiles = table.newAppend();
+    files.forEach(appendFiles::appendFile);
+    appendFiles.commit();
+  }
+
+  private List<DataFile> newFiles(int numFiles, long sizeInBytes) {
+    List<DataFile> files = Lists.newArrayList();
+    for (int fileNum = 0; fileNum < numFiles; fileNum++) {
+      files.add(newFile(sizeInBytes));
+    }
+    return files;
+  }
+
+  private DataFile newFile(long sizeInBytes) {
+    return DataFiles.builder(PartitionSpec.unpartitioned())
+      .withPath("/path/to/data-a.parquet")
```

Comment:
Nit: reused file names aren't currently checked, but it still seems odd to reuse the same name. What about adding a UUID?